renderdoc/os/posix/linux/linux_process.cpp

```diff
@@ -35,6 +38,42 @@ char **GetCurrentEnvironment()
   return environ;
 }
 
+std::vector<int> getSockets(pid_t childPid)
+{
+  std::vector<int> sockets;
+  string dirPath = StringFormat::Fmt("/proc/%d/fd", (int)childPid);
+  DIR *pDir = opendir(dirPath.c_str());
+  if(pDir == NULL)
+  {
+    RDCWARN("Cannot open directory %s", dirPath);
+    return sockets;
+  }
+  struct dirent *pDirent;
+  while((pDirent = readdir(pDir)) != NULL)
+  {
+    if(pDirent->d_type != DT_LNK)
+      continue;
+
+    std::stringstream target;
+    target << dirPath.c_str() << "/" << pDirent->d_name;
+
+    // the link target if it is a socket is in the "socket:[%u]" format
+    char linkname[1024];
+    ssize_t length = readlink(target.str().c_str(), linkname, 1023);
+    if(length == -1)
+      continue;
+
+    linkname[length] = '\0';
+
+    uint inode;
```

Comment:
`uint32_t` rather than the non-standard `uint` here please, and initialise to 0 just to be save.

Reply:
I have modified my commit.